lib/src/parse/stylesheet.dart

```diff
@@ -2857,6 +2898,17 @@ abstract class StylesheetParser extends Parser {
             buffer.writeCharCode(scanner.readChar());
           }
 
+        case $lparen || $lbracket:
+          var bracket = scanner.readChar();
+          buffer.writeCharCode(bracket);
+          brackets.add(opposite(bracket));
+
+        case $rparen || $rbracket:
+          if (brackets.isEmpty) break loop;
```

Comment:
Given that we're now explicitly parsing matching brackets here, it's probably best to throw a meaningful error here.

Note that this is technically a breaking change whether or not we throw an error here: `[foo#{"]:is(bar"}) {x: y}` is currently valid, but will become an error here. I think it's vanishingly unlikely that anyone is doing that, though, so it should be fine to break, but we should add a "potentially breaking bug fix" note in the changelog about it.